src/js/tabs/firmware_flasher.js

```diff
@@ -587,6 +587,12 @@ TABS.firmware_flasher.initialize = function (callback) {
                 self.flashingMessage(i18n.getMessage('firmwareFlasherFirmwareLocalLoaded', self.parsed_hex.bytes_total), self.FLASH_MESSAGE_TYPES.NEUTRAL);
             }
         }
+        function checkAsciiLimits(input) {
+            for (let i=0; i < input.length; i++) {
+                if (input.charCodeAt(i) > 127) { return false; }
```

Comment:
Are you sure that it is necessary to block 8 bit ASCII (up to 255)?

Reply:
Well, what do we treat the higher values as? My ASCII chart from the MSDOS manual has different mapping than [ISO-8859-1](https://en.wikipedia.org/wiki/ISO-8859-1). And [UTF-8](https://en.wikipedia.org/wiki/UTF-8#Description) wants values higher than 0x7F stored as multiple bytes. `reader.readAsText(file);` further down parses the input as UTF-8 by default.

Limiting to 127 prevents any issues, but maybe we could just say that 0-255 is unicode 0-255 and its not UTF-8 encoded? We might want to come back later and use UTF-8 if someone makes a better OSD with emoji support though.

Reply:
I'd convert them to ISO-8859-1 or ISO-8859-15 with transliteration. While not perfect, this will avoid throwing unhelpful errors for things like comments in non-English, which in my opinion is better than telling users they can't do something - the only thing we have to fend against is JavaScript unhelpfully 'converting' UTF-8 by masking the higher bits.

Reply:
I'll see what I can find for transliteration. hopefully its built into javascript. If we can read UTF-8, and transliterate down to ISO-8859-15, that result just needs to be fed to the part that really wants an array of numbers from 0-255

Reply:
@Docteh: Yeah, not sure what can be used. `iconv` and friends seems to be the gold standard, but a quick glance did not turn up anything that is browser ready, which is what we need if we want it to work in the Chrome web app.

Reply:
@mikeller I looked at iconv-lite which is all in javascript, I did one test and it seems to just replace unknown characters with question mark. I was expecting that it would see fancy quotes and make them the boring ones. 

How about, read through character by character, keep an eye out for the first # in a line, if we see something we don't like before the # mark, `error`, otherwise replace with a question mark. 
Putting an explanation in GUI.log would also help.

`# In one of those southern whalesmen, on a long three or four years’`
`# In one of those southern whalesmen, on a long three or four years?`

Actually for the context of comments, maybe a different replacement character is in order, like underscore. 
`# In one of those southern whalesmen, on a long three or four years_`

Reply:
Lol, good approach to transliteration that `iconv-lite` has.

The approach of erroring in non-comments sounds reasonable.

I'm wondering if there is a way to distinguish between 8 bit ASCII and > 8 bit unicode, so we can allow users to use all available OSD symbols in things like the craft name?

Reply:
I did some 'sperimentin:

```
console.log("a".charCodeAt(0))
=> 97
console.log("à".charCodeAt(0))
=> 224
console.log("ā".charCodeAt(0))
=> 257
console.log("€".charCodeAt(0))
=> 8364
```

So it looks like characters that are part of ISO 8859-1 will be mapped to their codes <= 255, and will not pose problems due to being truncated into control characters - we can leave them in anywhere in the config.